### ISSUE TITLE
Hold Request - Submitting report with expenses that on hold returns unexpected error

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#37017]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
**Found when validating PR :** https://github.com/Expensify/App/pull/33897

**Version Number:** 1.4.43-6
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause-Internal Team
**Slack conversation:**

## Action Performed:
Precondition: create a workspace, and invite an employee and approver
1. Go to https://staging.new.expensify.com/

2. Log in as an employee
3. Navigate to a workspace chat
4. Create a manual request
5. Navigate to the expense detail page
6. Click on 3 dots > Hold request
7. Enter a reason and click on Hold request
8. Return to the expense report and submit the report

## Expected Result:
The report is submitted and the system message is displayed

## Actual Result:
The error message "Unexpected error, please try again later" is shown. After refreshing the page, the system message that the report was submitted is displayed.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/115492554/47079fd4-ad93-44cc-98b6-38433f34368c

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by izarutskaya** [comment]
We think that this bug might be related to #vip-bills
CC @davidcardoza

**Comment by amyevans** [comment]
I can repro on the latest `main` Auth/Web/App locally, this is the SubmitReport response I receive:
```
{
    "jsonCode": 666,
    "title": "Cannot complete action",
    "message": "You do not have the permission to do the requested action.",
    "requestID": "pxZ0B3",
    "onyxData": []
}
```

I tried searching for logs from the QA tester to confirm it was the same error but can't find them 🤔 (searched with applausetester+0219pm1a@applause.expensifail.com / reportID 4675844924345473).

Since it's a backend error it likely shouldn't block App deploy, but I'm also not entirely sure of expected behavior so perhaps this action shouldn't be allowed in the first place.

cc @robertjchen

**Comment by robertjchen** [comment]
Hm, not a Deploy Blocker per se, since we're working on the approval/payment flow: https://expensify.slack.com/archives/C02NZ2HGJAZ/p1708478069824719

That said, does it still break things if removed from Hold? 🤔

**Comment by amyevans** [comment]
> That said, does submit still break if the request is NOT on/removed from hold? 🤔

Yeah, if I unhold the request and then submit, it works (I did have to disable the `getAllowedWriteCommandsCount` from interfering locally fwiw).

**Comment by amyevans** [comment]
Going to unassign since I'm OOO next week

**Comment by mallenexpensify** [comment]
@robertjchen @amyevans should this be internal or external? Bumped back to `Daily` since it's a bug we want to fix. Adding to  #wave-collect too.

**Comment by robertjchen** [comment]
Internal- we will revisit after Partial Pay/Approve is out since it relates to that flow.

**Comment by mallenexpensify** [comment]
Bobby Bottles, is there a link for `Partial Pay/Approve ` ?

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by mallenexpensify** [comment]
Tried to repro but not seeing HOLD option in upper tight when I click the three stacked dots
<img width="721" alt="image" src="https://github.com/Expensify/App/assets/22508304/2ab276cb-57cd-49a6-a408-1759e3c7efc9">

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by mallenexpensify** [comment]
@allroundexperts can you reproduce? If so, can you provide some deets, especially regarding how to put an expense on hold? If you're unable to repro, I'll close.  Thx

**Comment by allroundexperts** [comment]
I do see the hold option. Make sure that you have multiple manual requests in a IOU report and then open any of the request. Clicking on the `three dots` in the header does show up the hold option. However, after pressing it and trying to submit the report, I can not reproduce the error mentioned in the issue description.

**Comment by allroundexperts** [comment]
Ref:

https://github.com/Expensify/App/assets/30054992/97eca095-45a1-4dbf-a793-f9ae63381836

**Comment by mallenexpensify** [comment]
Thanks @allroundexperts , following your steps I was able to use `Hold`.  Was also unable to reproduce, so closing.
